messaging/models.py

```diff
@@ -25,37 +25,8 @@ def five(self):
 		# and as always Python handles negatives nicely
 		return self.get_messages().reverse()[:5][::-1]
 
-		"""The following code was too beautiful to delete.
-	# @param *pals_str list of usernames (strings) of involved members
-	# @return the messagethread containing exactly these users (created if none exists)
-	def get_thread_from_str(*pals_str):
-		members = (Member.objects.get(equiv_user__username=x) for x in pals_str)
-		return MessageThread.get_thread(*members)
-
-	# retrieve the single MessageThread containing exactly this set of users
-	# function can take list of Members as arguments, or as an iterable collection, because Python
-	def get_thread(*participants_exactly):
-		from django.db.models import Count
-
-		# Selects the threads with the correct number of participants
-		query = MessageThread.objects.annotate(num_participants=Count('participants')).filter(num_participants=len(participants_exactly))
-
-		# make sure each participant is in the set
-		for x in participants_exactly:
-			query = query.filter(participants=x)
-
-		# we now have a strong enough condition:
-		# if we have exactly the right number of people,
-		# and no person looked for is absent,
-		# then therefore the list of present is exactly the list of people looked for!
-
-		thread, created = query.get_or_create()
-		# so if after all that, no thread exists, we need to make it and add people in
-		if created:
-			for x in participants_exactly:
-				thread.participants.add(x)
-
-		return thread"""
+	# There was some nice code written by Ash that was "too beautiful to delete".
```

Comment:
So it was neat because it uniquely identified a message thread by its set of users, and it was an issue I'd faced in SQL right beforehand. It's essentially the divide operator from relational algebra! It works quite neatly in Django, i thought.

Of course, there isn't any point keeping it around, since it was no longer needed x)

Reply:
:heavy_division_sign: